File: frontend/src/types/album.ts
// アルバム関連の型定義（OpenAPI仕様準拠）

export interface Album {
  id: string // UUID
  title: string
  description: string
  creator: string // UUID
  images: string[] // 画像UUIDの配列
  created_at: string // ISO 8601 format
  updated_at: string // ISO 8601 format
}

export interface AlbumItem {
  id: string // UUID
  title: string
  creator: string // UUID
}

export interface CreateAlbumRequest {
  title: string
  description: string
  images?: string[] // 画像UUIDの配列
}

export interface UpdateAlbumRequest {
  title?: string
  description?: string
  images?: string[] // 画像UUIDの配列
}

// クエリパラメータ
export interface GetAlbumsParams {
  creator_id?: string // UUID
  before_date?: string // ISO 8601 date-time
  after_date?: string // ISO 8601 date-time
  limit?: number // 1-100, default: 20
  offset?: number // minimum: 0, default: 0
}

File: backend/internal/domain/album.go
package domain

import (
	"time"

	"github.com/google/uuid"
)

// Album represents an album entity in the domain
type Album struct {
	Id          uuid.UUID   `json:"id"`
	Title       string      `json:"title"`
	Description string      `json:"description"`
	Creator     string      `json:"creator"`
	Images      []uuid.UUID `json:"images"`
	CreatedAt   time.Time   `json:"created_at"`
	UpdatedAt   time.Time   `json:"updated_at"`
}

// AlbumItem represents a simplified album item for list views
type AlbumItem struct {
	Id      uuid.UUID `json:"id"`
	Title   string    `json:"title"`
	Creator string    `json:"creator"`
}

// AlbumFilter represents filtering criteria for albums
type AlbumFilter struct {
	CreatorID  *string
	BeforeDate *time.Time // Filter by created_at
	AfterDate  *time.Time // Filter by created_at
	Limit      *int
	Offset     *int
	//あとはIsFavorite(*bool)とか？
}

// PostAlbumParams represents parameters for creating a new album
type PostAlbumParams struct {
	Title       string      `json:"title"`
	Description string      `json:"description"`
	Creator     string      `json:"creator"`
	Images      []uuid.UUID `json:"images"`
}

// UpdateAlbumParams represents parameters for updating an album
type UpdateAlbumParams struct {
	Title       *string      `json:"title,omitempty"`
	Description *string      `json:"description,omitempty"`
	Images      *[]uuid.UUID `json:"images,omitempty"`
}
